### ISSUE TITLE
Cast error on init

### ISSUE THREAD

**Issue by guiguito** [riccardo-lomazzi/camera_macos#2]
I've tested the library on a project of mine and I had the following error:

```
The following _CastError was thrown building FutureBuilder<CameraMacOSArguments?>(dirty, dependencies: [MediaQuery], state: _FutureBuilderState<CameraMacOSArguments?>#6a26b):
Null check operator used on a null value

The relevant error-causing widget was: 
  FutureBuilder<CameraMacOSArguments?> FutureBuilder:file:///Users/guiguito/dev/flutter/.pub-cache/hosted/pub.dartlang.org/camera_macos-0.0.5/lib/camera_macos.dart:56:12
When the exception was thrown, this was the stack: 
#0      CameraMacOSViewState.build.<anonymous closure> (package:camera_macos/camera_macos.dart:96:67)
#1      _FutureBuilderState.build (package:flutter/src/widgets/async.dart:616:55)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:4992:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4878:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5050:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:4604:5)
#6      StatefulElement.update (package:flutter/src/widgets/framework.dart:5082:5)
#7      Element.updateChild (package:flutter/src/widgets/framework.dart:3570:15)

```

I tried in a project from scratch with the code from your example and I get the same error. Any idea?
I took care of added the relevant keys in the plist l (NSCameraUsageDescription and NSMicrophoneUsageDescription) as well as the *.entitlements files.

I'm using macOS 13.0 ventura. 
The webcam is working well anywhere else.

**Comment by guiguito** [comment]
I found the root cause. It works on a standard macOS laptop webcam.
But it does not work on a USB plugged webcam.


The call to the initialize method returns : {error: {message: Could not find a suitable camera on this device, details: null, code: CAMERA_INITIALIZATION_ERROR}}

Is it somthing fixable?

**Comment by riccardo-lomazzi** [comment]
Hi @guiguito, sorry for the late response.

Currently there's no official support for external camera devices, meaning that the package wasn't coded with those in mind.
I'll update the README to specify it.
 
I need to test this missing feature with the relative equipment, once I'll get a hold on that, then I'll update the package.

**Comment by riccardo-lomazzi** [comment]
I've managed to recover an old hd camera that can act as a webcam. 
I can record videos, though I'm working on the audio part.
I've also reworked the initialization method, so that you can choose a particular device to inizialize the `CameraMacOSView` widget with.

You can track the progress on the `external_camera_support` branch. ([Link](https://github.com/riccardo-lomazzi/camera_macos/tree/external_camera_support))

**Comment by riccardo-lomazzi** [comment]
Hi @guiguito , I've just published the `0.0.6` version of the package with new features and support for external devices.
Hope it serves you well.
I'll close this issue for now.
